### PR TITLE
CRUD API for collection and activity models

### PR DESCRIPTION
**Pull request by idegtiarov**
Implement API view sets for Activities and Collections

Add possibility to use CRUD operations for Activity and Collection models.

### CODE REVIEW

File: bridge_adaptivity/api/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        data = {
+            'name': name,
+            "metadata": None,
+            "strict_forward": True,
+            "owner": self.user.id
+        }
+        response = self.client.post(self.list_url, data)
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.data['name'], name)
+        deleted = self.client.delete(self.list_url + '{}/'.format(response.data['id']))
```
Comment (alexbojko) [inline bridge_adaptivity/api/tests/test_api.py]: should be better to rewrite 
`"{}{}/".format(self.list_url, response.data['id'])`

Comment (alexbojko) [reply bridge_adaptivity/api/tests/test_api.py]: or build reverse ?

File: bridge_adaptivity/api/tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        self.assertEqual(response.data['name'], self.name)
+        activities = self.client.get(self.list_url).data
+        self.assertEqual(len(activities), 2)
+
+    @patch('module.tasks.sync_collection_engines.apply_async')
+    def test_delete_activity(self, mock_apply_async):
+        response = self.client.post(self.list_url, self.data)
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.data['name'], self.name)
+        deleted = self.client.delete(self.list_url + '{}/'.format(response.data['id']))
```
Comment (alexbojko) [inline bridge_adaptivity/api/tests/test_api.py]: ditto

File: bridge_adaptivity/api/serializers.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+class ActivitySerializer(serializers.ModelSerializer):
+    """
+    Serializer for Activity API view set.
+
+    Serializer is based on the Activity model's fields.
+    """
+
+    class Meta:
+        model = Activity
+        fields = '__all__'
```
Comment (alexbojko) [inline bridge_adaptivity/api/serializers.py]: is it secure to allow to serialize all fields?

File: bridge_adaptivity/api/serializers.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+class CollectionSerializer(serializers.ModelSerializer):
+    """
+    Serializer for Collection API view set.
+
+    Serializer is based on the Collection model's fields.
+    """
+
+    class Meta:
+        model = Collection
+        fields = '__all__'
```
Comment (alexbojko) [inline bridge_adaptivity/api/serializers.py]: is it secure?

File: bridge_adaptivity/config/settings/base.py [hunk 5]
```diff
@@ -143,3 +145,16 @@
...
+REST_FRAMEWORK = {
+    'DEFAULT_AUTHENTICATION_CLASSES': (
+        'rest_framework.authentication.TokenAuthentication',
+        # for browsable api view usage
+        'rest_framework.authentication.SessionAuthentication',
+    ),
+    'DEFAULT_PERMISSION_CLASSES': (
+        'rest_framework.permissions.IsAuthenticated',
+    ),
+    'TEST_REQUEST_DEFAULT_FORMAT': 'json'
```
Comment (alexbojko) [inline bridge_adaptivity/config/settings/base.py]: why not to add comma in the end of this line?

File: bridge_adaptivity/api/serializers.py (hunk 3 above)
Comment (idegtiarov) [reply bridge_adaptivity/api/serializers.py]: We have not any sensitive data in the Activity model.

File: bridge_adaptivity/api/serializers.py (hunk 4 above)
Comment (idegtiarov) [reply bridge_adaptivity/api/serializers.py]: ditto

File: bridge_adaptivity/config/settings/base.py (hunk 5 above)
Comment (idegtiarov) [reply bridge_adaptivity/config/settings/base.py]: 👍 nice catch

File: bridge_adaptivity/api/tests/test_api.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    @classmethod
+    @patch('module.tasks.sync_collection_engines.apply_async')
+    def setUpTestData(cls, mock_apply_async):
+        cls.user = BridgeUser.objects.create(
+            username='test',
+            password='test',
+        )
+        cls.token = Token.objects.create(user=cls.user)
+        cls.collection = Collection.objects.create(name='col1', owner=cls.user)
+        cls.activity = Activity.objects.create(name='act1', collection=cls.collection, tags='test')
```
[the remark this answers is not in the document]
Comment (idegtiarov) [reply bridge_adaptivity/api/tests/test_api.py]: Thanks.

File: bridge_adaptivity/api/tests/test_api.py (hunk 1 above)
Comment (idegtiarov) [reply bridge_adaptivity/api/tests/test_api.py]: `reverse` seems to be the best choice. 
done.

File: bridge_adaptivity/api/tests/test_api.py (hunk 2 above)
Comment (idegtiarov) [reply bridge_adaptivity/api/tests/test_api.py]: done

**Pull request merged**